src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java

```diff
@@ -472,7 +519,71 @@ private QueryOuterClass.Query getVectorSearchQuery(List<BuiltCondition> conditio
           .build();
     }
   }
-
+  /**
+   * A separate method to build vector search query by using expression, expression can contain
+   * logic operations like 'or','and'..
+   */
+  private QueryOuterClass.Query getVectorSearchQueryByExpression(
+      Expression<BuiltCondition> expression) {
+    QueryOuterClass.Query builtQuery = null;
+    if (projection().doIncludeSimilarityScore()) {
+      switch (commandContext().similarityFunction()) {
+        case COSINE, UNDEFINED -> {
+          return new QueryBuilder()
+              .select()
+              .column(ReadType.DOCUMENT == readType ? documentColumns : documentKeyColumns)
+              .similarityCosine(
+                  DocumentConstants.Fields.VECTOR_SEARCH_INDEX_COLUMN_NAME,
+                  CustomValueSerializers.getVectorValue(vector()))
+              .from(commandContext.namespace(), commandContext.collection())
+              .where(expression)
+              .limit(limit)
+              .vsearch(DocumentConstants.Fields.VECTOR_SEARCH_INDEX_COLUMN_NAME)
+              .build();
+        }
+        case EUCLIDEAN -> {
+          return new QueryBuilder()
+              .select()
+              .column(ReadType.DOCUMENT == readType ? documentColumns : documentKeyColumns)
+              .similarityEuclidean(
+                  DocumentConstants.Fields.VECTOR_SEARCH_INDEX_COLUMN_NAME,
+                  CustomValueSerializers.getVectorValue(vector()))
+              .from(commandContext.namespace(), commandContext.collection())
+              .where(expression)
+              .limit(limit)
+              .vsearch(DocumentConstants.Fields.VECTOR_SEARCH_INDEX_COLUMN_NAME)
+              .build();
+        }
+        case DOT_PRODUCT -> {
+          return new QueryBuilder()
+              .select()
+              .column(ReadType.DOCUMENT == readType ? documentColumns : documentKeyColumns)
+              .similarityDotProduct(
+                  DocumentConstants.Fields.VECTOR_SEARCH_INDEX_COLUMN_NAME,
+                  CustomValueSerializers.getVectorValue(vector()))
+              .from(commandContext.namespace(), commandContext.collection())
+              .where(expression)
+              .limit(limit)
+              .vsearch(DocumentConstants.Fields.VECTOR_SEARCH_INDEX_COLUMN_NAME)
+              .build();
+        }
+        default -> {
+          throw new JsonApiException(
+              ErrorCode.VECTOR_SEARCH_INVALID_FUCTION_NAME,
+              ErrorCode.VECTOR_SEARCH_INVALID_FUCTION_NAME.getMessage());
```

Comment:
Append name of function in question; otherwise caller doesn't know offending value?

Reply:
solved